pkg/flowaggregator/flowaggregator.go

```diff
@@ -724,3 +904,101 @@ func (fa *flowAggregator) createInfoElementForTemplateSet(ieName string, enterpr
 	}
 	return ie, nil
 }
+
+func (fa *flowAggregator) watchConfiguration(stopCh <-chan struct{}) {
+	klog.Info("Watching for FlowAggregator configuration file")
+	for {
+		select {
+		case <-stopCh:
+			return
+		case event, ok := <-fa.configWatcher.Events:
+			if err := fa.handleWatcherEvent(event, ok); err != nil {
+				// if the watcher cannot add mounted configMap or the configMap is not readable, we kill the flow-aggregator pod (serious error)
+				klog.Fatalf("Cannot watch or read configMap: %v", err)
+			}
+		case err, _ := <-fa.configWatcher.Errors:
+			if err != nil {
+				// if the error happens to watcher, we kill the flow-aggregator pod.
+				// watcher might be shut-down or broken in this situation.
+				klog.Fatalf("configWatcher err: %v", err)
+			}
+		}
+	}
+}
+
+func (fa *flowAggregator) handleWatcherEvent(event fsnotify.Event, ok bool) error {
+	if !ok {
```

Comment:
I don't understand why you made `ok` a parameter, instead of simply not calling the function if the watch is closed

Reply:
I was thinking if there is a case that would close the watcher channel. If the channel is closed for some reason, I think we should kill the fa Pod as well?

Reply:
That doesn't answer my question. If the channel is closed, you can kill the Pod but there is no reason to handle this case in this function instead of in `watchConfiguration` directly. This function shouldn't be called at all if the channel has been closed.

Reply:
You're right. I've updated it and kill the fa Pod when watcher channel is somehow closed.